src/Storages/RabbitMQ/StorageRabbitMQ.cpp

```diff
@@ -151,24 +391,56 @@ Pipe StorageRabbitMQ::read(
     if (num_created_consumers == 0)
         return {};
 
+    auto sample_block = metadata_snapshot->getSampleBlockForColumns(column_names, getVirtuals(), getStorageID());
+
+    auto new_context = std::make_shared<Context>(context);
+    if (!schema_name.empty())
+        new_context->setSetting("format_schema", schema_name);
```

Comment:
Why do we need to share context in each `read`? Actually here just created another shared ptr, all changes will be reflected in the original object.